[hotfix] Fix db2 Db2Dialect error

andyphone opened the pull request:
Removed the semicolon from the SQL query as it caused a syntax error . ( Caused by: com.ibm.db2.jcc.am.SqlSyntaxErrorException: DB2 SQL Error: SQLCODE=-104, SQLSTATE=42601, SQLERRMC=;;columnA, SOURCE.columnB);END-OF-STATEMENT, DRIVER=4.26.14 ) This error only occurs when a primary key is defined in the table structure.

andyphone committed:
Update Db2Dialect.java

Removed the semicolon from the SQL query as it caused a syntax error . ( Caused by: com.ibm.db2.jcc.am.SqlSyntaxErrorException: DB2 SQL Error: SQLCODE=-104, SQLSTATE=42601, SQLERRMC=;;columnA, SOURCE.columnB);END-OF-STATEMENT, DRIVER=4.26.14 )

eskabetxe commented:
Thanks for the improvement @andyphone 
Could you add some test to validate that we don't introduce this again. 
Thanks

RocMarshal reviewed (commented):
Thanks @andyphone for the pr and @eskabetxe @davidradl for the review.

@davidradl May I know the latest new progress about adding unit test cases for the pr?

andyphone committed:
Update Db2PreparedStatementTest.java

### Motivation

The SQL generated by `Db2Dialect#getUpsertStatement` previously ended with a trailing semicolon (`;`).

When executing the generated statement through the DB2 JDBC driver, the trailing semicolon may cause a syntax error. In particular, when the target table contains a primary key, the driver may throw the following exception:

```
com.ibm.db2.jcc.am.SqlSyntaxErrorException:
DB2 SQL Error: SQLCODE=-104, SQLSTATE=42601
```

This happens because DB2 JDBC does not expect a semicolon at the end of a SQL statement executed through `PreparedStatement`.

### Changes

This PR removes the trailing semicolon from the SQL statement generated by `Db2Dialect#getUpsertStatement`.

As a result, the generated `MERGE INTO` statement can be executed correctly by the DB2 JDBC driver.

### Tests

The existing `Db2DialectTest#testUpsertStatement` test has been updated to reflect the corrected SQL format (without the trailing semicolon).
This ensures that a trailing semicolon will not be reintroduced in future changes.

### Impact

This change only affects the SQL formatting of the DB2 dialect and does not affect other JDBC dialect implementations.

andyphone commented:
@davidradl plz have a look am i wrong

andyphone commented:
I've added the test update and adjusted the expected SQL accordingly.
Could you please take another look when you have time? @eskabetxe

RocMarshal commented:
Hi， @andyphone Thanks for the update.
Pls run `mvn spotless:apply` in the repo dir to fix the codestyle issue before the next push.

RocMarshal commented:
Since this PR has gone unanswered for a long time, I reopened it to push it forward while preserving the original author’s sign-off.
